src/checks/y_check_db_access_in_ut.clas.abap

```diff
@@ -51,131 +86,185 @@ CLASS y_check_db_access_in_ut IMPLEMENTATION.
   ENDMETHOD.
 
 
-  METHOD inspect_statements.
-    inspect_class_definition( structure ).
+  METHOD inspect_tokens.
+    CASE structure-stmnt_type.
+      WHEN scan_struc_stmnt_type-class_definition.
+        add_line_to_defined_classes( statement = statement
+                                     structure = structure ).
+
+      WHEN scan_struc_stmnt_type-class_implementation.
+        check_class( index = index
+                     statement = statement
+                     structure = structure ).
+
+    ENDCASE.
+  ENDMETHOD.
+
+
+  METHOD is_persistent_object.
+    cl_abap_structdescr=>describe_by_name( EXPORTING p_name = obj_name
+                                           EXCEPTIONS OTHERS = 1 ).
+    result = xsdbool( sy-subrc = 0 ).
+  ENDMETHOD.
 
-    IF has_framework = abap_true.
+
+  METHOD is_internal_table.
+    DATA(second_token) = get_token_abs( statement-from + 1 ).
+    DATA(third_token) = get_token_abs( statement-from + 2 ).
+
+    IF second_token = 'INTO'.
       RETURN.
     ENDIF.
 
-    super->inspect_statements( structure ).
+    DATA(table) = COND #( WHEN second_token = 'FROM' THEN third_token
+                                                     ELSE second_token ).
+
+    result = xsdbool( is_persistent_object( table ) = abap_false ).
   ENDMETHOD.
 
 
-  METHOD inspect_tokens.
-    DATA(tokens) = consolidade_tokens( statement ).
+  METHOD add_line_to_defined_classes.
+    DATA(class_config) = VALUE properties( name = get_class_name( structure ) ).
 
-    LOOP AT tokens_not_allowed ASSIGNING FIELD-SYMBOL(<token_not_allowed>).
-      IF tokens NP <token_not_allowed>.
-        CONTINUE.
-      ENDIF.
+    IF line_exists( excepted_classes[ name = class_config-name ] )
+    OR line_exists( defined_classes[ name = class_config-name ] ).
+      RETURN.
+    ENDIF.
 
-      IF is_an_attribution( statement ) = abap_true.
-        CONTINUE.
-      ENDIF.
+    class_config-risk_level = get_risk_level( statement ).
 
-      IF has_ddic_itab_same_syntax( <token_not_allowed> ) = abap_true
-      AND is_internal_table( statement ) = abap_true.
-        CONTINUE.
-      ENDIF.
+    IF is_part_of_framework( structure ) = abap_true.
+      APPEND class_config TO excepted_classes.
+    ELSE.
+      APPEND class_config TO defined_classes.
+    ENDIF.
+  ENDMETHOD.
 
-      DATA(check_configuration) = detect_check_configuration( statement ).
 
-      IF check_configuration IS INITIAL.
-        RETURN.
-      ENDIF.
+  METHOD check_class.
+    IF NOT line_exists( defined_classes[ name = get_class_name( structure ) ] ).
+      RETURN.
+    ENDIF.
 
-      raise_error( statement_level     = statement-level
-                    statement_index     = index
-                    statement_from      = statement-from
-                    error_priority      = check_configuration-prio ).
+    DATA(forbidden_tokens) = get_forbidden_tokens( get_class_name( structure ) ).
```

Comment:
store class name in a variable to avoid multiple calls to the method that determines it